braviarc/braviarc.py

```diff
@@ -146,6 +146,23 @@ def send_command(self, command):
         """Sends a command to the TV."""
         self.send_req_ircc(self.get_command_code(command))
 
+    def get_source(self, source):
+        """Returns list of Sources"""
+        original_content_list = []
+        stIdx = 0
+        while True:
```

Comment:
As we don't have any clue on how many channels, the only way is infinite loop and stop when I found an array of length 0.

If we know how to get the amount of channels on a previous API call, than we could only loop the necessary.